src/libraries/Microsoft.PowerFx.LanguageServerProtocol/Handlers/LanguageServerOperationContext.cs

```diff
@@ -77,10 +77,13 @@ private IPowerFxScope GetScope(string uri)
         /// <param name="uri">Uri to use for scope creation.</param>
         /// <param name="factory">Nl Handler Factory.</param>
         /// <param name="nLParams">Ml Params.</param>
+        /// <param name="cancellationToken"> Cancellation Token. </param>
         /// <returns>NlHandler Instance.</returns>
-        internal NLHandler GetNLHandler(string uri, INLHandlerFactory factory, BaseNLParams nLParams = null)
+        internal Task<NLHandler> GetNLHandler(string uri, INLHandlerFactory factory, BaseNLParams nLParams = null, CancellationToken cancellationToken = default)

```

Comment:
Suggestion: rename to `GetNLHandlerAsync` since this is now an asynchronous method.

Reply:
+1 
This is actually a very important convention.

Reply:
agreed. just forgot about adding async there